backend/src/database.cpp

```diff
@@ -0,0 +1,294 @@
+    std::string query = "SELECT COUNT(*) FROM Recordings WHERE id = '" + uuid + "' AND file_path = '" + file_path + "';";
+    sqlite3_stmt *stmt;
+    int rc = sqlite3_prepare_v2(db, query.c_str(), -1, &stmt, nullptr);
+    if (rc != SQLITE_OK) {
+        std::cerr << "Failed to execute query: " << sqlite3_errmsg(db) << std::endl;
+        return false;
+    }
+
+    bool exists = false;
+    if (sqlite3_step(stmt) == SQLITE_ROW) {
+        exists = sqlite3_column_int(stmt, 0) > 0;
+    }
+
+    sqlite3_finalize(stmt);
+    return exists;
+}
+
+
+//VENDORS
+bool Database::insert_vendor(const std::string &oid, const std::string &name) {
+    std::string check_query = "SELECT COUNT(*) FROM Vendors WHERE oid = '" + oid + "';";
+    sqlite3_stmt *stmt;
+    int rc = sqlite3_prepare_v2(db, check_query.c_str(), -1, &stmt, nullptr);
+    if (rc != SQLITE_OK) {
+        std::cerr << "Failed to execute query: " << sqlite3_errmsg(db) << std::endl;
+        return false;
+    }
+    bool exists = false;
+    if (sqlite3_step(stmt) == SQLITE_ROW) {
+        exists = sqlite3_column_int(stmt, 0) > 0;
+    }
+    sqlite3_finalize(stmt);
+    if (exists) {
+        return true; //vendor already exists
+    }
+
+    std::string query = "INSERT INTO Vendors (oid, name) VALUES ('" + oid + "', '" + name + "');";
+    return execute_query(query);
+}
+
+std::vector<std::vector<std::string>> Database::get_vendors() {
+    std::string query = "SELECT * FROM Vendors;";
+    return select_query(query);
+}
+
+//NETWORKS
+bool Database::insert_network(const std::string &ssid, const std::string &bssid, const std::string &psk, uint32_t total_packet_count, uint32_t decrypted_packet_count, uint32_t group_packet_count, const std::string &security, const std::string recording_id, uint32_t group_rekeys, const std::string &vendor_oid) {
+    //check if the network with the BSSID already exists
+    std::string check_query = "SELECT COUNT(*) FROM Networks WHERE bssid = '" + bssid + "';";
+    sqlite3_stmt *stmt;
+    int rc = sqlite3_prepare_v2(db, check_query.c_str(), -1, &stmt, nullptr);
+    if (rc != SQLITE_OK) {
+        std::cerr << "Failed to execute query: " << sqlite3_errmsg(db) << std::endl;
+        return false;
+    }
+    bool exists = false;
+    if (sqlite3_step(stmt) == SQLITE_ROW) {
+        exists = sqlite3_column_int(stmt, 0) > 0;
+    }
+    sqlite3_finalize(stmt);
+
+    if (exists) {
+        return true;
+    }
+
+    std::string query = "INSERT INTO Networks (ssid, bssid, psk, total_packet_count, decrypted_packet_count, group_packet_count, security, recording_id, group_rekeys, vendor_oid) VALUES ('" + ssid + "', '" + bssid + "', '" + psk + "', " + std::to_string(total_packet_count) + ", " + std::to_string(decrypted_packet_count) + ", " + std::to_string(group_packet_count) + ", '" + security + "', '" + recording_id + "', " + std::to_string(group_rekeys) + ", '" + vendor_oid + "');";
```

Comment:
Think about moving this monstrosity into a prepared stmt with ? signs and filling at query prep time